packages/paste-core/components/button/src/index.tsx

```diff
@@ -42,23 +49,34 @@ const getButtonState = ({loading, disabled}: ButtonProps): ButtonStates => {
   return 'default';
 };
 
-const Button: React.FC<ButtonProps> = props => {
-  const buttonState = getButtonState(props);
+const Button: React.FC<ButtonProps> = ({
+  as,
+  children,
+  disabled,
+  fullWidth,
+  href,
+  loading,
+  size,
+  tabIndex,
+  variant,
+  ...props
+}) => {
+  const buttonState = getButtonState(disabled, loading);
   const showLoading = buttonState === 'loading';
-  const disabled = buttonState !== 'default';
+  const showDisabled = buttonState !== 'default';
```

Comment:
Keeping this as `disabled` would have been fine, it's less about showing disabled, but actually setting the disabled attribute

Reply:
Right. I renamed that because of the conflict with the destructured disabled prop, but since we're not passing that I'll change it back.